src/tools/rbtree.c

```diff
@@ -1199,6 +1194,27 @@ static void print_rbnode(const rbnode *node, unsigned depth, uintptr_t minstart,
     }
     printf_log(LOG_NONE, ")");
 }
+int cache_check(const rbtree_t *tree){
+    if (!tree->root) return 0;
+    //find right most
+    rbnode *right_node = tree->root;
+    while (right_node) {
+        if(!right_node->right)
+            break;
+        right_node = right_node->right;
+    }
+    if(tree->righter != right_node)
+        printf_log(LOG_NONE, "<invalid rightmost node> ");
+    rbnode *left_node = tree->root;
+    while (left_node) {
+        if(!left_node->left)
+            break;
+        left_node = left_node->left;
+    }
+    if(tree->lefter != left_node)
+        printf_log(LOG_NONE, "<invalid leftmost node> ");
+    return 1;
```

Comment:
This function does not return whether the leftmost and rightmost nodes are valid.